protocol/group/mutableautoselect_test.go

```
@@ -1250,19 +1252,66 @@ func TestSetURLOverrides_ClearsPersistedEntryOnChange(t *testing.T) {
+
+	s, obs := newTestMUR(t, "dead", "live")
 	s.defaultURL = "http://probe.test/"
-	obs["a"].On("DialContext").Return(nil, errors.New("dial denied"))
-	obs["b"].On("DialContext").Return(nil, errors.New("dial denied"))
+	s.cfg.ladderTotalBudget = 10 * time.Second
+	recordSuccess(s, "dead", 100)
+	recordSuccess(s, "live", 200)
+
+	obs["dead"].On("DialContext").Return(nil, errors.New("dial denied"))
+	obs["live"].dial = func() (net.Conn, error) {
+		return net.Dial("tcp", srv.Listener.Addr().String())
```

golangci-lint (2.12.2)

[error] net.Dial must not be called. use (*net.Dialer).DialContext

(noctx)